### PR TITLE
Capture environment details when running the test suite

### LINKED ISSUE #19: Include PHP module and system utility versions in the test run report
To have better visibility into what's causing failures, it'd be helpful to include PHP module and system utility versions in the test run report.

Specifically:

* [x] ImageMagick

Potentially others listed in https://github.com/ataylorme/WordPress-Hosting-Best-Practices-Documentation/blob/ee171bfac07dd595fcc75e40c5ddadfda8477748/server-environment/server-environment.md#php-extensions

### PR DESCRIPTION
**Pull request by danielbachhuber**
Fixes #19

### CODE REVIEW

**Comment by danielbachhuber** [comment]
@octalmage Produces output like this:

```
$ cat wp-test-runner/tests/phpunit/build/logs/env.json
{
    "php_version": "5.6.29",
    "php_modules": {
        "imagick": "3.4.3",
        "filter": "0.11.0",
        "xml": false,
        "pcre": false,
        "mod_xml": false,
        "bcmath": false
    },
    "system_utils": {
        "imagemagick": "6.7.7-10",
        "ghostscript": "9.05"
    }
}
```

**Comment by danielbachhuber** [comment]
For the record, I put the system reporting mechanism in `wp-tests-config.php` so we didn't have to make an extra SSH call in `test.php`. It will run automatically when the test suite is run.

File: prepare.php [hunk 1]
```diff
@@ -38,12 +38,44 @@
...
+foreach( \$php_modules as \$php_module ) {
+	\$env['php_modules'][ \$php_module ] = phpversion( \$php_module );
+}
+\$ret = shell_exec( 'convert --version' );
+preg_match( '#Version: ImageMagick ([^\s]+)#', \$ret, \$matches );
+\$env['system_utils']['imagemagick'] = isset( \$matches[1] ) ? \$matches[1] : false;
+\$env['system_utils']['ghostscript'] = shell_exec( 'gs --version' );
+file_put_contents( __DIR__ . '/tests/phpunit/build/logs/env.json', json_encode( \$env, JSON_PRETTY_PRINT ) );
+EOT;
+$logger_replace_string = '// wordpress/wp-config.php will be ignored.' . PHP_EOL;
```
Comment (octalmage) [inline prepare.php]: Would this cause an issue if it's ran on Windows and the wp-tests-config has a unix newline?

Comment (danielbachhuber) [inline prepare.php]: Only if the file was _saved_ on Windows. Is this a use-case we need to solve for?

Comment (octalmage) [inline prepare.php]: probably not! Lets get this merged so I can pull the changes into the report branch.

**Pull request merged**